packages/ra-core/src/types.ts

```diff
@@ -4,6 +4,10 @@ export interface Record {
     [key: string]: any;
 }
 
+export interface RecordMap {
+    [id: string]: Record;
```

Comment:
unfortunately, there is a bug in TypeScript that doesn't let you use Identifer as a key type, but I'm afraid this will break if the id is a number